services/web3id-issuer/src/main.rs

```diff
@@ -268,13 +300,43 @@ async fn issue_credential(
         amount: Amount::zero(),
     };
 
+    let holder_id = request
+        .credential_subject
+        .id
+        .ty
+        .extract_public_key()
+        .ok_or(Error::InvalidId)?
+        .into();
+    if request.credential_subject.id.network != state.network {
+        return Err(Error::InvalidId);
+    }
+    let valid_from = Timestamp::from_timestamp_millis(
+        u64::try_from(request.valid_from.timestamp_millis())
+            .map_err(|_| Error::InvalidTimeRange)?,
+    );
+    let valid_until = request
+        .valid_until
+        .map(|v| {
+            u64::try_from(v.timestamp_millis())
+                .map_err(|_| Error::InvalidTimeRange)
+                .map(Timestamp::from_timestamp_millis)
+        })
+        .transpose()?;
```

Comment:
Maybe it also makes sense to validate that `valid_from` < `valid_until`?  The registry allows it, but it will never be activated, because it expires before that. I think it makes sense to fail early.

Reply:
Added this check.